### PR TITLE
[CIS GCP] Aggregate Log sinks for evaluation

### PR DESCRIPTION
**Pull request by uri-weisman**
### Summary of your changes
A new Logging fetcher is created to gather log sink resources for evaluation.
Currently, the fetcher is set to operate only when we assess a GCP Organization, extending its support to project scope would demand extra permissions during onboarding, allowing only Organization Admins to onboard projects.

### Related Issues
- Related: https://github.com/elastic/csp-security-policies/pull/309

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: resources/providers/gcplib/inventory/provider.go [hunk 1]
```diff
@@ -460,11 +493,23 @@ func getEcsGcpCloudData(ctx context.Context, crm *ResourceManagerWrapper, keys G
...
 func getProjectId(ancestors []string) string {
-	return strings.Split(ancestors[0], "/")[1]
+	parts := strings.Split(ancestors[0], "/") // projects/1234567890
+
+	if parts[0] == "projects" {
+		return parts[1]
+	}
+
+	return ""
 }
```
Comment (uri-weisman) [inline resources/providers/gcplib/inventory/provider.go:513]: updated in order to be aligned with resources that are being created at the folder/org level.
Their ancestors will look like `[folders/123, org/123]`

File: resources/providers/gcplib/inventory/provider.go [hunk 2]
```diff
@@ -460,11 +493,23 @@ func getEcsGcpCloudData(ctx context.Context, crm *ResourceManagerWrapper, keys G
...
 	last := ancestors[len(ancestors)-1]
-	return strings.Split(last, "/")[1]
+	parts := strings.Split(last, "/") // organizations/1234567890
+
+	if parts[0] == "organizations" {
+		return parts[1]
+	}
+
+	return ""
 }
```
Comment (orouz) [inline resources/providers/gcplib/inventory/provider.go:503]: this change was done to account for the case where there is only 1 ancestor - the org?

Comment (uri-weisman) [reply resources/providers/gcplib/inventory/provider.go:503]: It was done to account for a project without an organizational association.

File: resources/fetching/fetcher.go [hunk 3]
```diff
@@ -53,11 +53,13 @@ const (
 	AccessAnalyzers           = "aws-access-analyzers"
 
 	GcpMonitoringType = "gcp-monitoring"
+	GcpLoggingType    = "gcp-logging"
 	GcpServiceUsage   = "gcp-service-usage"
 
 	CloudIdentity          = "identity-management"
 	CloudCompute           = "cloud-compute"
 	MonitoringIdentity     = "monitoring"
+	LoggingIdentity        = "logging"
```
[the remark this answers is not in the document]
Comment (uri-weisman) [reply resources/fetching/fetcher.go:62]: All the rest of the types and subtypes are stored here.
The constants we keep in `asset.go` are resource types we collect using the asset inventory, the fact that we derive the subtype out of those types is only a bonus.

File: resources/fetching/fetchers/gcp/logging_fetcher.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+type GcpLoggingAsset struct {
+	Type    string
+	subType string
+
+	Asset *inventory.LoggingAsset `json:"asset,omitempty"`
+}
+
+func NewGcpLoggingFetcher(_ context.Context, log *logp.Logger, ch chan fetching.ResourceInfo, provider inventory.ServiceAPI) *GcpLoggingFetcher {
```
Comment (orouz) [inline resources/fetching/fetchers/gcp/logging_fetcher.go:44]: logging fetcher is kinda broad. rule 2.1 is about logging too. maybe include the name sinks to be more specific

**Review by orouz — APPROVED** [review]

**Pull request merged**